File: Codes/p15/p15.m
close all; clear all; clc;
path(path,genpath('../p13'));
x       = [7, 38, 4, 23, 18];
x2      = x.^2;
x2mu    = compMean(x2);
xmu     = compMean(x);
xvar    = compVariance(x);
fprintf('The Variance of X \t= %.2f \n', xvar);
fprintf('E[X^2]-E[X]^2 \t\t= %.2f \n', x2mu - xmu^2);

File: python修改张铁海/p15.py
import numpy as np
from p13 import compMean
from p13 import compVariance

x= [7, 38, 4, 23, 18]
x=np.mat(x)
x2= np.square(x)
# print(x2)
x2mu= compMean.compMean(x2)
xmu = compMean.compMean(x)
xvar= compVariance.compVariance(x)
print('The Variance of X \t= %.2f \n'%xvar)
print('E[X^2]-E[X]^2 \t\t= %.2f \n'%(x2mu - np.square(xmu)))